src/schema/firefox-schemas-import.js

```diff
@@ -473,47 +482,88 @@ export function foldSchemas(schemas) {
   return foldedSchemas;
 }
 
+// Normalize the JSONSchema, gets an array of JSONSchema objects and return an array of
+// normalized JSONSchema objects (a JSONSchema file may contain more than one API namespaces,
```

Comment:
```suggestion
// normalized JSONSchema objects (a JSONSchema file may contain more than one API namespace,
```

Reply:
Does the normalization return an array with each item being one API namespace?

Reply:
yes, that's what this method is expected to do.

to be precise:
- the version without this changes was returning the normalized version of the JSONSchema object **not related to extending the manifest** (which,  if there is one, is expected to be an API)
- the new version does basically the same thing but it does not assume that there will be only one API namespace per file, because that's not completely true (for menus.json and browser_action.json at the moment), + the addition of normalizing the $import explicitly

Given that I'm going to still do some other tweaks based on your review comments, I'm going to take another look to this method to see if there is any other unexpected corner case that we may want to throw an explicit error for.

Reply:
In the end I haven't added an additional check for this, we are currently assuming that in each JSONSchema file, after filtering out any schema entry related to the manifest namespace, all the remaining entries should be API namespaces, and that should still be a reasonable assumption.

We can re-evaluate later on if we determine that we should be more strict (especially given that at that point it may be more clear what to check to decide if the schema data is not the one we expect).